SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/DispatcherTest.cpp

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include <memory>
+
+#include "ComponentTestUtils.h"
+#include "SpatialView/Callbacks.h"
+#include "SpatialView/ComponentData.h"
+#include "SpatialView/Dispatcher.h"
+#include "SpatialView/EntityDelta.h"
+#include "SpatialView/EntityView.h"
+#include "SpatialView/OpList/EntityComponentOpList.h"
+#include "SpatialView/ViewDelta.h"
+#include "SpatialViewUtils.h"
+#include "Tests/TestDefinitions.h"
+
+#define DISPATCHER_TEST(TestName) GDK_TEST(Core, Dispatcher, TestName)
+
+namespace
+{
+constexpr Worker_ComponentId COMPONENT_ID = 1000;
+constexpr Worker_ComponentId OTHER_COMPONENT_ID = 1001;
+constexpr Worker_EntityId ENTITY_ID = 1;
+constexpr Worker_EntityId OTHER_ENTITY_ID = 2;
+constexpr double COMPONENT_VALUE = 3;
+constexpr double OTHER_COMPONENT_VALUE = 4;
+
+void PopulateViewDeltaWithComponentAdded(SpatialGDK::ViewDelta& Delta, SpatialGDK::EntityView& View, const Worker_EntityId EntityId,
+										 const Worker_ComponentId ComponentId, const double Value)
+{
+	SpatialGDK::EntityComponentOpListBuilder OpListBuilder = SpatialGDK::EntityComponentOpListBuilder{};
+	OpListBuilder.AddComponent(EntityId, SpatialGDK::CreateTestComponentData(ComponentId, Value));
+	SetFromOpList(Delta, View, OpListBuilder);
```

Comment:
Ah possible driveby I didn't notice from before. This function is transferring ownership so it should take the builder by value. Then it should be moved from here.

Reply:
I don't think that's possible, iirc I had to pass it by reference as the copy constructor wasn't available for all fields. but maybe there is another way to do it?

Reply:
moving means it isn't being copied- do you mean the oplistbuilder can't be copied? Because then we could force that it has to be moved in instead.

Reply:
Made the argument to the function an &&

Reply:
and undone